test/PaymasterCommitment.test.ts

```diff
@@ -189,10 +189,83 @@ contract('Paymaster Commitment', function ([_, relayOwner, relayManager, relayWo
       assert.closeTo(paid, gasUsed, 50)
     })
 
+    it('paymaster should not pay for requests exceeding msg.data size limit', async () => {
+      const r = await makeRequest(web3, {
+        request: {
+          // nonce: '4',
+          data: recipientContract.contract.methods.emitMessage('').encodeABI()
+        },
+        relayData: { paymaster }
+
+      }, sharedRelayRequestData, chainId, forwarderInstance)
+      const gasAndDataLimits = await paymasterContract.getGasAndDataLimits()
+      const hugeApprovalData = '0x' + 'ef'.repeat(parseInt(gasAndDataLimits.calldataSizeLimit))
+      await expectRevert(
+        relayHubInstance.relayCall(10e6, r.req, r.sig, hugeApprovalData, externalGasLimit, {
+          from: relayWorker,
+          gas: externalGasLimit,
+          gasPrice
+        }), 'msg.data exceeded limit'
+      )
+    })
+
+    it('paymaster should not pay for requests with max msg.data size if it rejects in pre', async () => {
+      await paymasterContract.setRevertPreRelayCall(true)
+      const r = await makeRequest(web3, {
+        request: {
+          // nonce: '4',
+          data: recipientContract.contract.methods.emitMessage('').encodeABI()
+        },
+        relayData: { paymaster }
+
+      }, sharedRelayRequestData, chainId, forwarderInstance)
+
+      const gasAndDataLimits = await paymasterContract.getGasAndDataLimits()
+      const hugeApprovalData = '0x' + 'ef'.repeat(parseInt(gasAndDataLimits.calldataSizeLimit) - 1000)
+      const relayCallParams: [number, RelayRequest, string, string, number, Truffle.TransactionDetails?] = [10e6, r.req, r.sig, hugeApprovalData, externalGasLimit]
+      const method = relayHubInstance.contract.methods.relayCall(...relayCallParams)
+      assert.equal(gasAndDataLimits.calldataSizeLimit, toBuffer(method.encodeABI()).length.toString(),
+        'relayCall() msg.data should be set to max size')
+      const txdetails: Truffle.TransactionDetails = {
+        from: relayWorker,
+        gas: externalGasLimit,
+        gasPrice
+      }
+      relayCallParams.push(txdetails)
+      const res = await relayHubInstance.relayCall(...relayCallParams)
+      expectEvent(res, 'TransactionRejectedByPaymaster', { reason: encodeRevertReason('You asked me to revert, remember?') })
+    })
+
+    it('paymaster should not pay for requests with max msg.data size if it accepts in pre but forwarder fails', async () => {
```

Comment:
What is new in this test? How does it relate to the issue we are discussing?

Reply:
This is literally the attack raised by the audit.

Reply:
That is correct, I misread the sentence